src/pages/PositionApply.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { useEffect } from 'react';
+import { useState } from 'react';
+import { useContext } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useHistory, useParams } from 'react-router-dom';
+import { Button, ButtonToolbar, Notification, ControlLabel, Form, FormControl, FormGroup, Schema } from 'rsuite';
+import { getPosition, Position, PositionApplicationParams, submitPositionApplication } from '../utils/api/PositionApplication';
+import { Credentials } from '../utils/Authentication';
+import Loading from './Loading';
+
+function PositionApply() {
+  const [t] = useTranslation();
+  const history = useHistory();
+  let{positionId} = useParams<any>();
+  const [position, setPosition] = useState<Position>();
+  const ctx = useContext(Credentials);
+  //set up required variable for rsuite forms.
+  let form: any =  undefined;
+  const msg_required = "This field is required";
+  const model = Schema.Model({
+      personalStatement: Schema.Types.StringType()
+        .isRequired(msg_required)
+        .minLength(1,msg_required)
+  })
+  //form data setup
+  const [formData, setFormData] = useState<Record<string,any>>({
+      personalStatement:''
+  })
+  const [formErrors, setFormErrors] = useState<Record<string, any>>({});
+
+  useEffect(()=>{
+    //Return early if no context is provided.
+    if(!ctx || !ctx.credentials.authenticated) return;
+    //If logged in, attempt to get the position details
+    getPosition(positionId)
+        .then((res) => {
+            setPosition(res);
+        })
+        .catch((err)=>{
+            console.error(err)
+        })
+  },[]);
```

Comment:
This should be sensitive to `positionId`

Reply:
wym

Reply:
```
  },[]);
```

What does this do?

Reply:
as in put id in the [ ]

Reply:
The effect gonna execute every time the variable in the [] changes